IDR 0.4.4 merge branch

manics opened the pull request:
This will be split into separate PRs.
- [x] --depends-on https://github.com/openmicroscopy/ansible-role-jekyll-build/pull/9

--on-hold
--exclude

Main changes:
- `/shared` scratch space in jupyterhub notebooks
- Moved jekyll source dir to `/srv/www/src/` and added initial config to support https://github.com/IDR/idr.openmicroscopy.org/pull/24
- Prometheus monitoring for [postgres](https://github.com/openmicroscopy/ansible-role-prometheus-postgres)
- Bumped [Postgres ansible role to 3.0.0-m2](https://github.com/openmicroscopy/ansible-role-postgresql/pull/7), updated variables, enabled restricted option (omero-readonly requires TEMPORARY table privs).
- Experimental: Ensure journald logs are [kept after reboot](https://unix.stackexchange.com/a/159390), need to check for disk usage and/or log rotation
- Bumped to Kubernetes 1.7.4 (may contain deployment bugs which require deleting instead of updating a deployment when the config is changed)
- Enabled [anonymous grafana account](http://docs.grafana.org/installation/configuration/#auth-anonymous) for interactive viewing (can view over ssh tunnel without login, not safe for public proxying due to lack of protection for backend data: https://github.com/IDR/deployment/issues/49)
- Experimental: proxy [rendered pngs of grafana dashboards](http://docs.grafana.org/reference/sharing/#direct-link-rendered-image) with 60 second cache, e.g. http://example.org/grafana/idr-vertical (accepts size query parameters, need to check whether there are limits, hangs with invalid dashboard names).

snoopycrimecop commented:
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1553](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1553/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1553/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - ansible/requirements.yml

snoopycrimecop commented:
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1554](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1554/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1554/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - ansible/requirements.yml

snoopycrimecop commented:
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1555](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1555/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1555/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - ansible/requirements.yml

snoopycrimecop commented:
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1556](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1556/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1556/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - ansible/requirements.yml

snoopycrimecop commented:
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1557](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1557/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1557/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - ansible/requirements.yml

snoopycrimecop commented:
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1558](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1558/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1558/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - ansible/requirements.yml

manics commented:
This has been split into the following PRs:
- #55
- #56
- #57
- #58
- #59
- #60
- #61
- #62
- #63
- #64
- #65
- #66
- #67

sbesson commented:
Most open PRs targeted at IDR 0.4.4 have been integrated. Is this merge branch still necessary?

jburel commented:
Are the new files in this PR needed to be added to master?
ansible/experimental-omero-web-addons.yml
ansible/grafana-dashboards/idr-postgresql.json

What about the changes in ansible/idr-09-monitoring.yml and  ansible/management-grafana.yml
Do they need to be included?

manics commented:
Thanks for checking, I've rebased this and added the remaining commits to:
- https://github.com/IDR/deployment/pull/69
- https://github.com/IDR/deployment/pull/70

jburel commented:
General question: how do we know that there are not other commits used for deployment that have been forgotten?

manics commented:
Same as before, diff master against this branch?

jburel commented:
The question is more: do we have other ways to validate that the diff?
At the moment, there is none (which is fine) Trying to get my head around everything

sbesson commented:
Looking at https://github.com/manics/deployment/compare/idr-0-4-4...snoopycrimecop:master/merge/trigger?expand=1 is the extra `iviewer` section expected?

manics commented:
Isn't that in https://github.com/IDR/deployment/pull/70 ?

sbesson commented:
:+1: GitHub branch comparison API has limitations when comparing differences between 2 integrations branchess but a manual `git diff` shows the two branches have no differences.

sbesson commented:
All PRs have now been integrated into `master`. Closing this.
